test/es-module/test-vm-main-context-default-loader.js

```diff
@@ -0,0 +1,138 @@
+    let result = await script.runInThisContext();
+    assert.strictEqual(result.constants.F_OK, fs.constants.F_OK);
+
+    const imported = compileFunction('return import("fs")', [], options)();
+    result = await imported;
+    assert.strictEqual(result.constants.F_OK, fs.constants.F_OK);
+  }
+
+  const modulePath = fixtures.path('es-modules', 'message.mjs');
+  fs.copyFileSync(modulePath, tmpdir.resolve('message.mjs'));
+
+  {
+    const namespace = await import(modulePath);
+    const script = new Script('import("./message.mjs")', options);
+    const result = await script.runInThisContext();
+    assert.deepStrictEqual(result, namespace);
+  }
+
+  {
+    const namespace = await import(modulePath);
+    const imported = compileFunction('return import("./message.mjs")', [], options)();
+    const result = await imported;
+    assert.deepStrictEqual(result, namespace);
+  }
+}
+
+async function main() {
+  {
+    // Importing with absolute path as filename.
+    tmpdir.refresh();
+    const filename = tmpdir.resolve('index.js');
+    const options = {
+      filename,
+      importModuleDynamically: USE_MAIN_CONTEXT_DEFAULT_LOADER
+    };
+    await testNotFoundErrors(options);
+    await testLoader(options);
+  }
+
+  {
+    // Importing with file:// URL as filename.
+    tmpdir.refresh();
+    // We use a search parameter to bypass caching.
+    const filename = url.pathToFileURL(tmpdir.resolve('index.js')).href + '?t=1';
+    const options = {
+      filename,
+      importModuleDynamically: USE_MAIN_CONTEXT_DEFAULT_LOADER
+    };
+    await testNotFoundErrors(options);
+    await testLoader(options);
+  }
+
+  {
+    // For undefined or non-path/URL filenames, import() should resolve to the cwd.
+    tmpdir.refresh();
+    process.chdir(tmpdir.path);
+    const undefinedOptions = {
+      importModuleDynamically: USE_MAIN_CONTEXT_DEFAULT_LOADER
+    };
+    const nonPathOptions = {
+      filename: 'non-path',
+      importModuleDynamically: USE_MAIN_CONTEXT_DEFAULT_LOADER
+    };
+    // Run the error tests first to avoid caching.
+    await testNotFoundErrors(undefinedOptions);
+    await testNotFoundErrors(nonPathOptions);
+
+    // createContext() with null referrer also resolves to cwd.
+    {
+      const options = {
+        importModuleDynamically: USE_MAIN_CONTEXT_DEFAULT_LOADER,
+      };
+      const ctx = createContext({}, options);
+      const s = new Script('Promise.resolve("import(\'./message.mjs\')").then(eval)', {
+        importModuleDynamically: common.mustNotCall(),
+      });
+      await assert.rejects(s.runInContext(ctx), { code: 'ERR_MODULE_NOT_FOUND' });
+    }
+
+    await testLoader(undefinedOptions);
+    await testLoader(nonPathOptions);
+
+    {
+      const options = {
+        importModuleDynamically: USE_MAIN_CONTEXT_DEFAULT_LOADER,
+      };
+      const ctx = createContext({}, options);
+      const modulePath = fixtures.path('es-modules', 'message.mjs');
```

Comment:
An import specifier should be a URL string or a relative path string. This fails the windows test.

```suggestion
      const moduleUrl = fixtures.fileURL('es-modules', 'message.mjs');
```